File: python/tcp/cliente_tcp.py
import socket

# OFFSET pessoal (secao 3.3 do roteiro) - deve ser o MESMO do servidor
OFFSET = 81

HOST = "localhost"
PORTA = 5000 + OFFSET

with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as cliente:
    cliente.connect((HOST, PORTA))
    print(f"[TCP] Conectado ao servidor na porta {PORTA}.")
    print("[TCP] Digite 'hora' para pedir o horário ou 'sair' para encerrar.")

    arquivo = cliente.makefile("r")

    while True:
        mensagem = input("> ")
        cliente.sendall((mensagem + "\n").encode("utf-8"))
        print(arquivo.readline().strip())

        if mensagem.lower() == "sair":
            break

File: lab-redes/java/tcp/ClienteTCP.java
import java.io.*;
import java.net.*;

public class ClienteTCP {

    // OFFSET pessoal (secao 3.3 do roteiro) - deve ser o MESMO do servidor
    static final int OFFSET = 81;

    public static void main(String[] args) throws IOException {
        String host = "localhost";
        int porta = 5000 + OFFSET;

        try (Socket socket = new Socket(host, porta);
             PrintWriter saida = new PrintWriter(socket.getOutputStream(), true);
             BufferedReader entrada = new BufferedReader(
                     new InputStreamReader(socket.getInputStream()));
             BufferedReader teclado = new BufferedReader(new InputStreamReader(System.in))) {

            System.out.println("[TCP] Conectado ao servidor na porta " + porta + ".");
            System.out.println("[TCP] Digite 'hora' para pedir o horario ou 'sair' para encerrar.");

            String linha;
            while (true) {
                System.out.print("> ");
                linha = teclado.readLine();
                if (linha == null) break;

                saida.println(linha);
                System.out.println(entrada.readLine());

                if (linha.equalsIgnoreCase("sair")) break;
            }
        }
    }
}
